lib/AST/Expr.cpp

```diff
@@ -320,12 +320,12 @@ ConcreteDeclRef Expr::getReferencedDecl(bool stopAtParenExpr) const {
                  ->GetSubExpr()->getReferencedDecl(stopAtParenExpr)
 
   NO_REFERENCE(Error);
-  NO_REFERENCE(NilLiteral);
-  NO_REFERENCE(IntegerLiteral);
-  NO_REFERENCE(FloatLiteral);
-  NO_REFERENCE(BooleanLiteral);
-  NO_REFERENCE(StringLiteral);
-  NO_REFERENCE(InterpolatedStringLiteral);
+  SIMPLE_REFERENCE(NilLiteral, getInitializer);
+  SIMPLE_REFERENCE(IntegerLiteral, getInitializer);
+  SIMPLE_REFERENCE(FloatLiteral, getInitializer);
+  SIMPLE_REFERENCE(BooleanLiteral, getInitializer);
+  SIMPLE_REFERENCE(StringLiteral, getInitializer);
+  SIMPLE_REFERENCE(InterpolatedStringLiteral, getInitializer);
```

Comment:
Does this change cursor info at all? Could add some tests for that (though I'm not sure anyone would think to jump to definition on a literal 😅). It would also be a little weird for eg. the collection cases since I (think) only the first character (eg. `[`) would actually return a result.

Reply:
It looks like it doesn't; a quick check adding one of these to `SourceKit/CursorInfo/cursor_info_expressions` came back with `<empty cursor info; internal diagnostic: "Resolved to incomplete expression or statement.">` at the literal's location.